float64[] vertex_value
uint32[] edge_list